webpack/assets/javascripts/react_app/components/ColumnSelector/ColumnSelector.js

```diff
@@ -0,0 +1,200 @@
+import React, { useState } from 'react';
+import { PropTypes } from 'prop-types';
+import { Button, Modal, ModalVariant, TreeView } from '@patternfly/react-core';
+import { ColumnsIcon } from '@patternfly/react-icons';
+import { cloneDeep } from 'lodash';
+import { translate as __ } from '../../common/I18n';
+import API from '../../API';
+import { changeQuery } from '../../common/urlHelpers';
+import './column-selector.scss';
+
+const ColumnSelector = props => {
+  const {
+    data: { url, controller, categories, hasPreference },
+  } = props;
+
+  const initialColumns = cloneDeep(categories);
+  const [isModalOpen, setModalOpen] = useState(false);
+  const [selectedColumns, setSelectedColumns] = useState(categories);
+  const [tablePreference, setTablePreference] = useState(hasPreference);
```

Comment:
This is not needed, since we always get the right value from the server. We don't need to modify it either.
```suggestion
```